Improve look and feel of kubearchinspect repository (#17)
ArmDeveloperEcosystem/kubearchinspect: README.md

<!--
Copyright (C) 2024 Arm Limited or its affiliates and Contributors. All rights reserved.
SPDX-License-Identifier: Apache-2.0
-->
# KubeArchInspect

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
[![Main CI/CD](https://github.com/ArmDeveloperEcosystem/kubearchinspect/actions/workflows/main.yml/badge.svg)](https://github.com/ArmDeveloperEcosystem/kubearchinspect/actions/workflows/main.yml)

## Overview

Migrating your websites and services to run on Arm infrastructure can give multiple benefits in terms of cost and performance. The first phase in this process is to determine whether the container images in a Kubernetes cluster have Arm support, most do, but checking can be a manual and time consuming task. To make it easy we have developed the `kubearchinspect` tool which automates this process. If the version of one of your images lacks Arm support the tool will check for Arm architecture support in newer version and lets you know to update your image.

> This is an Open Source project and appreciate contributions and feedback, please see [Contributions Guidelines](CONTRIBUTING.md) for more information.

Reviewer: ```suggestion
This is Open Source Software and we appreciate contributions and feedback, please see the [Contribution Guidelines](CONTRIBUTING.md) for more information.
```

## Installation

> You can directly download the `kubearchinspect` executable from the [releases page](https://github.com/ArmDeveloperEcosystem/kubearchinspect/releases).

Reviewer: ```suggestion
Pre-built binaries are available from the [releases page](https://github.com/ArmDeveloperEcosystem/kubearchinspect/releases), alternatively see the [Development Guide](DEVELOPMENT.md) for building locally.
```

> **_NOTE:_** Kubernetes contexts can be shown using `kubectl config get-contexts` and set with `kubectl config set-context`.

Reviewer: Is having `kubectl` installed a pre-requisite? Could you give a worked example of how to get to the point where the tool can be run. Imagine someone is not that familiar with Kubernetes.

Key information is where the tool should be run, copied onto the target machine, run on your machine with kubectl` installed?

✅ registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.8.1
```

### Errors in Output

At present the tool will show an error if it cannot connect to the repository for an image for any reason.
This could include:
* Authentication failure
* Communication error
* Image no longer available

## Private Registry Authentication

> When `kubearchinspect` discovers an image from a registry that requires authentication, it uses the credential helper defined in `~/.docker/config.json` to obtain the required credentials. This is the same as the `docker` command line.

Reviewer: What is the same as the `docker` command line?

```suggestion
If `kubearchinspect` discovers an image from a registry that requires authentication, it uses the credential helper defined in `~/.docker/config.json` to obtain the required credentials. This is the same as the `docker` command line.
```

> ## Releases

Reviewer: I'm not sure what a product release is in this context, is it that pre-releases such as betas don't have a release note? Or would this be in the GitHub releases?
